### PR TITLE
Fix VTK Deprecation Warning in CMake

### PR DESCRIPTION
**Pull request by biergaizi**
This PR contains two changes.

## CMakeLists.txt: avoid deprecated names in VTK 9+.

In VTK 9+, components vtkIOXML, vtkIOGeometry, vtkIOLegacy, vtkIOPLY, etc, have been renamed to IOXML, IOGeometry, IOLegacy, IOPLY, etc. So use the new name if VTK 9+ is detected.

## CMakeLists.txt: don't call VTK_USE_FILE for 9.0+

The `VTK_USE_FILE` is no longer used starting with 8.90.

### CODE REVIEW

**Comment by thliebig** [comment]
Hmmpf, I should have not just reviewed but tested it. cmake does not like this for me... I will have a look

File: CMakeLists.txt [hunk 1]
```diff
@@ -134,13 +134,18 @@ message(STATUS "Boost_LIBRARIES: ${Boost_LIBRARIES}")
 INCLUDE_DIRECTORIES (${Boost_INCLUDE_DIRS})
 
 # vtk
-find_package(VTK COMPONENTS vtkIOXML vtkIOGeometry vtkIOLegacy vtkIOPLY NO_MODULE REQUIRED)
+find_package(VTK REQUIRED NO_MODULE)
```
Comment (thliebig) [inline CMakeLists.txt:137]: This line produces: 

> CMake Error at /usr/lib/cmake/vtk/VTK-vtk-module-find-packages.cmake:209 (find_package):
>   By not providing "FindCLI11.cmake" in CMAKE_MODULE_PATH this project has
>   asked CMake to find a package configuration file provided by "CLI11", but
>   CMake did not find one.
> 
>   Could not find a package configuration file provided by "CLI11" with any of
>   the following names:
> 
>     CLI11Config.cmake
>     cli11-config.cmake
> 
>   Add the installation prefix of "CLI11" to CMAKE_PREFIX_PATH or set
>   "CLI11_DIR" to a directory containing one of the above files.  If "CLI11"
>   provides a separate development package or SDK, be sure it has been
>   installed.
> Call Stack (most recent call first):
>   /usr/lib/cmake/vtk/vtk-config.cmake:166 (include)
>   CMakeLists.txt:137 (find_package)
> 

I may have to revert this changes? Sometimes cmake is just a PITA

**Comment by thliebig** [comment]
This is one of the reasons I try to never mess with cmake... it is just purely fractal and you just cannot test it on all possible machines and setups...

If anybode has a good idea how to replace cmake?

**Comment by thliebig** [comment]
`find_package(VTK)`

This line works for me....as soon as you add "required" nothing works anymore even though it is found just fine...
I HATE cmake and I do not know why vtk is such a pain with cmake?

**Comment by biergaizi** [comment]
Which system did you test this on? I might be able to look into it using a chroot container...

**Comment by thliebig** [comment]
Manjaro Linux... maybe we just use the "find_package(VTK)" and be done with it (for now)?

**Comment by biergaizi** [comment]
> Manjaro Linux

Aha, one of my test machine runs ArchLinux, and yes, I'm able to see the problem. Yes, just use `find_package(VTK)` for now should be okay. Are you going to commit it yourself, or should I open a new PR?

**Comment by thliebig** [comment]
No I just wanted to maybe confirm with you that the change is working for you too... I will fix it...
